libtransmission/net.h

```diff
@@ -75,6 +83,7 @@ struct tr_address
     {
         struct in6_addr addr6;
         struct in_addr addr4;
+        char unixSocketPath[TR_UNIX_ADDRSTRLEN];
```

Comment:
I don't love this but I'm not sure what to do about it. What used to be 16 bytes is now 109 bytes + padding, and `tr_address` is kept in peer-mgr's atom list for every known peer in every swarm. We're not going to have any unix socket path peers.

Is there a way to eliminate this cost or to separate it from the existing net.h code?

I hate suggesting this because it's trading one code smell for a slightly smaller code smell, but we could intern the socket path and make the field here a tr_quark. That's ugly too, but at least it won't increase the size of tr_address by 5x :slightly_smiling_face:

Reply:
Yes that is quite a huge cost, I didn't even think about that. The very easy (but stupid?) way of fixing this is to make it `char *unixSocketPath` and then just allocate memory for it. Then it would still be 16 bytes. But then of course you're risking leaks and such, and you'd have to be VERY sure there's no way a unix socket could slip through in `peer*.cc`.

Another, maybe less stupid, way of doing this would be to create a separate `tr_rpc_address` structure that's basically a clone of `tr_address` but with the extra part of the union for the path. This would require changing a few functions in `rpc-server.cc` and duplicating a slight amount of code like `tr_address_{to,from}_string`, but it would be the best solution IMO. If you think this approach sounds good I can start working on it.

Another advantage of the second method is that further refactoring wouldn't affect anything else `tr_address` uses. While implementing this I realized that eventually the goal of unix socket support in transmission's RPC would be to drop HTTP completely. That is a much larger refactor and there are still benefits that an http based unix socket has, but separating out `tr_address` from the rpc server's API would be a first step toward that.

Reply:
> If you think this approach sounds good I can start working on it.

Agreed, this option sounds the most promising right now.